homeassistant/components/cover/template.py

```diff
@@ -374,16 +392,28 @@ def async_update(self):
             except ValueError as ex:
                 _LOGGER.error(ex)
                 self._tilt_value = None
+
+        icon_picture_template = None
         if self._icon_template is not None:
+            property_name = 'icon'
+            icon_picture_template = self._icon_template
+        elif self._entity_picture_template is not None:
+            property_name = 'entity_picture'
+            icon_picture_template = self._entity_picture_template
+
+        if icon_picture_template is not None:
             try:
-                self._icon = self._icon_template.async_render()
+                setattr(self, '_{0}'.format(property_name),
+                        icon_picture_template.async_render())
```

Comment:
Simplify this. Maybe with a helper function. It's very hard to read. Or do it with a MAP helper and loop it.

Reply:
It seems pretty simple; unclear on your suggestions – could you share an example?

Reply:
@pvizeli After taking a look through, I'm still unclear on what exactly needs to change. Putting this into a helper function just abstracts the complexity; it doesn't remove it. Can you assist with further info?